pkg/log/logger.go

```diff
@@ -86,3 +88,38 @@ func newZapLoggerTo(destWriter io.Writer, level LogLevel, opts ...zap.Option) *z
 	return zap.New(zapcore.NewCore(&kube_log_zap.KubeAwareEncoder{Encoder: enc, Verbose: level == DebugLevel}, sink, lvl)).
 		WithOptions(opts...)
 }
+
+type ctxMarker struct{}
+
+type ctxLogger struct {
+	fields []interface{}
+}
+
+func WithFields(ctx context.Context, keysAndValues ...interface{}) context.Context {
```

Comment:
Is the idea to call this in `multitenant.WithTenant` ? If so maybe you should do it now to make this PR more understandable. In any case it's worth adding comments to both these functions

Reply:
addressed